### PR TITLE
Remove data races

### PR DESCRIPTION
**Pull request by fabxc**
@beorn7

### CODE REVIEW

File: dispatch.go [hunk 1]
```diff
@@ -273,6 +281,17 @@ func (ag *aggrGroup) String() string {
 	return fmt.Sprintf("%v", ag.fingerprint())
 }
 
+func (ag *aggrGroup) alertSlice() []*types.Alert {
+	ag.mtx.RLock()
+	defer ag.mtx.RUnlock()
+
+	var alerts []*types.Alert
```
Comment (beorn7) [inline dispatch.go]: Not that it matters a lot, but you could
```
alerts := make([]*types.Alert, 0, len(ag.alerts))
```
to avoid reallocations.

**Comment by beorn7** [comment]
:+1: (Feel free to disregard my comment if you prefer a few characters fewer over a few allocations fewer.)

File: dispatch.go (hunk 1 above)
Comment (beorn7) [inline dispatch.go]: Actually, the most idiomatic way is probably the following (in lieu of lines 288–292):

`return append(make([]*types.Alert, 0, len(ag.alerts)), ag.alerts...)`

**Comment by beorn7** [comment]
Please disregard my updated comment. :-)

File: dispatch.go (hunk 1 above)
Comment (fabxc) [inline dispatch.go]: `ag.alerts` is a map, not a slice. So this does not work.

**Comment by fabxc** [comment]
Merging for now, see comment.

**Pull request merged**